### ISSUE TITLE
[BUG]  Changes to SelectQueryHandler breaking the JSONAPI extension

### ISSUE THREAD

**Issue by Wartybliggens** [bolt/bolt#6115]
Since upgrading to version 3.2.3, the jsonapi extension is breaking. I realize this might not be the place to file a bug about that, however the jsonapi extension is an important one and you may be interested in case the fix belongs on the Bolt side. 

I filed a bug for the extension also:
https://github.com/xiaohutai/jsonapi/issues/34

Details
-------
 - Bolt Version: 3.2.3
 - Install type: Composer install
 - PHP version: 7.0
 - Used web server: Apache

Reproduction
------------
 1. **Bug summary**: 
    Using the jsonapi extension does not work due to a PHP exception.

 2. **Steps to reproduce**:
    * Install the jsonapi extension.
    * Add some content, for instance entries.
    * Go to .../json/entries

 4. **Expected result**: 
    You should be seeing json output, with a list of the entries.

 5. **Actual result**: 
    Instead you get an exception stemming from the new whitelisting code in SelectQueryHandler.php: 

Uncaught Exception: ContextErrorException
ContextErrorException in SelectQueryHandler.php line 74: 
Warning: preg_split() expects parameter 2 to be string, object given

```php
$metadata = $repo->getClassMetadata();
        $allowedParams = array_keys($metadata->getFieldMappings());
        $cleanParams = [];
        foreach ($queryParams as $fieldSelect => $valueSelect) {
            $stack = preg_split('/ *(\|\|\|) */', $fieldSelect);
            $valueStack = preg_split('/ *(\|\|\|) */', $valueSelect);
```

While debugging I found that "$valueSelect" is coming in as a "Bolt\Extension\Bolt\JsonApi\Converter\Parameter\Type\Page Object" instead of as a string.

**Comment by Wartybliggens** [comment]
Hello @skoop , @xiaohutai , @GawainLynch -  thank you for looking at this. I'm wondering, were the changes you made meant to fix the behavior of the json extension, or were they more about internals? For me, the extension still breaks but in a new way:
* There is no error now.
* The extension claims there are no items of the given content type (when of course there are). I tried with the built-in content types and my own custom one.

So I was wondering whether in fact the extension works for you. For now I've had to revert to an earlier version of Bolt in our use of it, which I would rather not do of course.

**Comment by GawainLynch** [comment]
> were the changes you made meant to fix the behavior of the json extension, or were they more about internals? 

Should have been both. @Raistlfiren did you also have a look at the extension side?

**Comment by Raistlfiren** [comment]
@Wartybliggens This should be fixed now in v3.0.7 of the JSONAPI extension. Please test it out and see if it works for you.

**Comment by Wartybliggens** [comment]
@Raistlfiren - thank you!
It is now fixed. One thing of note however: there are many PHP notices now for this extension when in debug mode. I will note that in the jsonapi bug.
Thanks again.
